File: libraryFinal/src/main/java/com/library/View/Menu.java
package com.library.View;

import java.util.Scanner;

public class Menu {

    ProductRegistrationView prv = new ProductRegistrationView();
    
    boolean exit;

    public void runMenu(){
        printHeader();
        while(!exit){
            printMainMenu();
            int choice = getInput();
            performAction(choice);
        }
    }

    private void printHeader(){
        System.out.println("+----------------------+");
        System.out.println("|      Welcome to      |");
        System.out.println("|       Library        |");
        System.out.println("+----------------------+");
    }

    private void printMainMenu(){
        System.out.println("\nPlease make a selection: ");
        System.out.println("1 - Register a book");
        System.out.println("2 - List the avaliable books");
        System.out.println("3 - Open your cart");
        System.out.println("4 - Checkout");
        System.out.println("--");
    }

    private int getInput(){
        Scanner sc = new Scanner(System.in);
        int choice = -1;
        while(choice < 0 || choice > 4){
            try {
                System.out.print("\nEnter your choice: ");
                choice = sc.nextInt();

            } catch (NumberFormatException e) {
                System.out.println("Invalid selection, please try again");
            }
        }
        return choice;
    }

    private void performAction(int choice){
        switch(choice){
            case 1:
                prv.runRegMenu();
                runMenu();
                //break;
            case 2:
                prv.retriveProdList();
                runMenu();
                //break;
            case 3:
                
                //break;
            case 4:
                //break;
            default:
            System.out.println("Invalid");
                //break;
        }
    }   
}

File: libraryCSharp/View/Menu.cs
using System;
using System.Runtime.InteropServices;
using libraryCSharp.Controller;

namespace libraryCSharp.View
{
    public class Menu
    {
        ProductRegistrationView prv;
        bool exit;

        public Menu()
        {
            prv = new ProductRegistrationView();
        }

        public void runMenu(){
            printHeader();
            while(!exit){
                printMainMenu();
                int choice = getInput();
                performAction(choice);
            }
        }
        
        private void printHeader(){
            Console.WriteLine("+----------------------+");
            Console.WriteLine("|      Welcome to      |");
            Console.WriteLine("|        Library       |");
            Console.WriteLine("+----------------------+");
        }
        
        private void printMainMenu(){
            Console.WriteLine("\nPlease make a selection: ");
            Console.WriteLine("1 - Register a products");
            Console.WriteLine("2 - List the avaliable products");
            Console.WriteLine("3 - Open your cart");
            Console.WriteLine("4 - Checkout");
            Console.WriteLine("--");
        }
        
        private int getInput(){
            int choice = -1;
            while(choice < 0 || choice > 4){
                try {
                    Console.WriteLine("\nEnter your choice: ");
                    choice = int.Parse(Console.ReadLine());

                } catch (FormatException e) {
                    Console.WriteLine("Invalid selection, please try again");
                }
            }
            return choice;
        }
        private void performAction(int choice){
            switch(choice){
                case 1:
                    prv.runRegMenu();
                    runMenu();
                break;
                case 2:
                    prv.RetriveProdList();
                    runMenu();
                break;
                case 3:
                
                break;
                case 4:
                break;
                default:
                    Console.WriteLine("Invalid");
                break;
            }
        }  
        
    }
}